### ISSUE TITLE
Forever pending on Safari 14.1.1

### ISSUE THREAD

**Issue by Polm90** [localForage/localForage#1022]
Hi,
I've a strange bug on Safari 14.1.1 (no problems on previous Safari versions or on other browsers): after I've done localforage.config({name: "app_name", key:"app_key_storage"}), i do a localforage.getItem("item_key").then(elem => console.log(elem)); but in most of case the promise is never solved and the app is  stuck forever. No errors (even if i add a catch), nothing.
But some other times, it works with no issues. It is not deterministic.

Any ideas about what could be wrong?

Thank you

**Comment by tofumatt** [comment]
I'm not yet sure what the cause would be, but I wonder if this is related to behaviours seen in #1004.

Eg: Are you only experiencing this with "get" calls or also with "set" calls?

**Comment by Polm90** [comment]
I think the bug is not related to localforage library but to [this](https://bugs.webkit.org/show_bug.cgi?id=226547) Safari bug.

Using [this](https://bugs.webkit.org/show_bug.cgi?id=226547#c26) provided workaround seems to work

**Comment by tofumatt** [comment]
For a moment I thought that too, but you said it's happening in iOS 14.1.1, which I thought wasn't affected by this bug. 🤔

**Comment by Polm90** [comment]
@tofumatt I said **Safari** 14.1.1 (on Mac) not **iOS** 14.1.1 ;)

**Comment by tofumatt** [comment]
Oh shoot! My bad 🤦🏻 

In that case, yes, I imagine that's exactly the issue.

**Comment by BuZZ-dEE** [comment]
https://github.com/jakearchibald/safari-14-idb-fix

**Comment by tofumatt** [comment]
Awesome, thanks for that link. I'll see about getting that integrated into the library this weekend 👍🏻

**Comment by Polm90** [comment]
@tofumatt please read [my comment](https://bugs.webkit.org/show_bug.cgi?id=226547#c37) on the bug track: it can be usefull if you're going to integrate the workaround in the localforage library

**Comment by BuZZ-dEE** [comment]
@tofumatt weekend is over. results? ;)

**Comment by dheimoz** [comment]
Hey @BuZZ-dEE , PR are always welcome and I believe speed up the process.

**Comment by tofumatt** [comment]
Indeed, I'm a bit behind but I do plan on looking this week. But a PR is totally welcome! 🙂

**Comment by BuZZ-dEE** [comment]
@tofumatt is `isIndexedDBValid.js` the correct place to use the [safari-14-idb-fix](https://github.com/localForage/localForage/issues/1022#issuecomment-863935406)

**Comment by Polm90** [comment]
Look also at [this comment](https://bugs.webkit.org/show_bug.cgi?id=226547#c44). I don't know if it's possible to integrate that workaround too...

**Comment by BRadHoc** [comment]
I'm still getting this bug on iOS 15 beta 2 which lead me to attempt a workaround to force localforage to use the LocalStorage driver, so it's probably going to be present in 14.7

**Comment by Polm90** [comment]
@BRadHoc do you still experience the bug using the safari-14-idb-fix?

**Comment by BRadHoc** [comment]
> @BRadHoc do you still experience the bug using the safari-14-idb-fix?

So far no, this is one that seems to happen randomly a different times though (I think depending on memory available) I'll keep testing.

**Comment by BRadHoc** [comment]
After some testing, the bug does still happen (however less frequently). I've narrowed it down to happening when you navigate between pages very quickly. Sometimes it just hangs and the promise never gets called.

**Comment by Polm90** [comment]
> After some testing, the bug does still happen (however less frequently). I've narrowed it down to happening when you navigate between pages very quickly. Sometimes it just hangs and the promise never gets called.

I was experiencing that problem too. And I've workarounded embedding the localforage call inside a window.requestAnimation as I described here https://bugs.webkit.org/show_bug.cgi?id=226547#c44 
Have you already tried?
